code/modules/projectiles/guns/energy/kinetic_accelerator.dm

```diff
@@ -75,6 +75,25 @@
 	holds_charge = TRUE
 	unique_frequency = TRUE
 	max_mod_capacity = 80
+	var/emagged = FALSE //an override for emagged mining borg PKAs to make them deal full damage in pressurized areas
+
+/obj/item/gun/energy/kinetic_accelerator/cyborg/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
+	if(iscyborg(user))
+		var/mob/living/silicon/robot/R = user
+		if(R.emagged)
+			emagged = TRUE //for emagged borgs only
+		else
+			emagged = FALSE //if you became un-emagged somehow, you don't get to use the emagged cyborg version of this gun
+	else
+		emagged = FALSE //if you're a human who's gotten ahold of a KA from an emagged cyborg somehow, you don't get to use the emagged cyborg version of this gun
```

Comment:
Same here, don't check every time you fire.

Reply:
I couldn't find anywhere else to add an emag check, and emagged mediborg defibs seem to do something like this.

Where should I add the emag check?

Reply:
`/mob/living/silicon/robot/proc/SetEmagged(new_state)`
In here you'd check if you have the `kinetic_accelerator` and modify it.

And where you gain the `kinetic_accelerator` you'd check if you are emagged and set up appropriately.